tools/Azure.Mcp.Tools.AzureManagedLustre/src/Commands/FileSystem/FileSystemUpdateCommand.cs

```diff
@@ -0,0 +1,140 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Azure.Mcp.Core.Commands;
+using Azure.Mcp.Core.Extensions;
+using Azure.Mcp.Core.Models.Option;
+using Azure.Mcp.Tools.AzureManagedLustre.Options;
+using Azure.Mcp.Tools.AzureManagedLustre.Options.FileSystem;
+using Azure.Mcp.Tools.AzureManagedLustre.Services;
+using Microsoft.Extensions.Logging;
+
+namespace Azure.Mcp.Tools.AzureManagedLustre.Commands.FileSystem;
+
+public sealed class FileSystemUpdateCommand(ILogger<FileSystemUpdateCommand> logger)
+    : BaseAzureManagedLustreCommand<FileSystemUpdateOptions>(logger)
+{
+    private const string CommandTitle = "Update Azure Managed Lustre FileSystem";
+
+    private new readonly ILogger<FileSystemUpdateCommand> _logger = logger;
+
+    public override string Name => "update";
+
+    public override string Description =>
+        """
+        Update maintenance window and/or root squash settings of an existing Azure Managed Lustre (AMLFS) file system. Provide either maintenance-day/time or root squash fields (no-squash-nid-list, squash-uid, squash-gid). Root squash fields must be provided if root squash is not None must be provided together.
+        """;
+
+    public override string Title => CommandTitle;
+
+    public override ToolMetadata Metadata => new()
+    {
+        Destructive = false,
```

Comment:
The current guidance is that most update operations are destructive. Given this has a substantial number of updates (based on the number of parameters) I would consider setting this to `true`.